### ISSUE TITLE
ANCM should forward the mapped Identity instead of the client certificate after IIS Client Cert Mapping Auth

### ISSUE THREAD

**Issue by mertozturk80** [dotnet/aspnetcore#42896]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

We are trying to use IIS Client Certificate mapping authentication together with ASP.NET Out of process handler. We see that ANCM module add MS-ASPNETCORE-CLIENTCERT  header to pass the certificate blob to the core process, but this prevent the target asp.net core process to lose the mapped user context. 

In other words: with IIS Client cert mapping(authmap.dll), IIS maps the certificate into a Windows identity. For the mapping to work IIS should forward the identity handle, not the certificate. The certificate no longer matters at that point, and aspnet core doesn’t know about or have the ability to take a client certificate and map it to a user. That doesn’t appear to be happening in ANCM, the IIS integration piece.

I know that after a successful certificate mapping , the User Context and the cached token is at the same place (HTTP Context), so hopefully it wont be a difficult change:
https://github.com/dotnet/aspnetcore/blob/52eff90fbcfca39b7eb58baad597df6a99a542b0/src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/forwardinghandler.cpp#L814

Thanks,
Mert

### Describe the solution you'd like

We would like to add IIS Certificate Mapping Authentication Integration to ANCM, via forwarding the identity handled instead of cert blob.

**Comment by blowdart** [comment]
I think we might want to do both, pass the certificate and the windows identity. It'd be unexpected to not have access to the certificate as well.

**Comment by adityamandaleeka** [comment]
@mertozturk80 Does using in-process hosting avoid this issue?

**Comment by mertozturk80** [comment]
@adityamandaleeka , no it does not avoid the issue.

**Comment by mertozturk80** [comment]
@blowdart, yes, but from the IIS integration perspective: passing the client cert i think have lower priority (nice to have), since its more important to pass the mapped identity. If we make sure that there is a IIS certificate mapped "user context", certificate have not much importance from that point on. Can be good for proxy scenarios may be?

**Comment by maras28** [comment]
I did the testing in my environment with an API app (.NET 6.0) hosted in IIS with both in-process and out-of-process cases. I was really hoping that in-process would work. However, even with in-process, the request results in 401 Unauthorized. I have the FREP logs. Client cert mapping auth maps the incoming cert to a Windows user but still the security context does not appear to be set on the application side. See the screenshot below.
![image](https://user-images.githubusercontent.com/1925572/180964422-ddec672b-9c05-494b-afaa-f8e4485268dc.png)

**Comment by blowdart** [comment]
@maras28 Yea it's not a supported configuration yet. Usually this setup has been for impersonation purposes, and we've moved on from inpersonation because more folks got it wrong than right, and with more modern auth it's not been needed.

@adityamandaleeka I think it's too late to get this in 7, but could come as a new feature somewhere between 7 & 8, as ancm is stand alone?

**Comment by adityamandaleeka** [comment]
@blowdart Any reason why it shouldn't just happen in 8?

**Comment by blowdart** [comment]
It's stand alone like ancm, so 🤷‍♂️ If folks are happen to wait then sure

**Comment by halter73** [comment]
What do you mean by "it's stand alone"?

**Comment by maras28** [comment]
If this, ANCM with this feature, could be released sooner, it would really help with a customer scenario in a hybrid Azure scenario where Azure API Management Self Hosted Gateway could authenticate against an existing on-prem API secured with Windows Authentication without code changes but with only IIS side configuration change (Client Certificate Authentication mapping).

**Comment by blowdart** [comment]
@halter73 in theory it shouldn't need any extra work in asp.net middleware. In theory :)

**Comment by Tratcher** [comment]
@blowdart The affected in-proc code is probably in the part of ANCM that ships with AspNetCore, not in the stand-alone shim. The shim is mainly responsible for finding the runtime, not processing requests. Out-of-proc doesn't have that separation and could apply to earlier versions, but we still only ship new ANCMs with new versions of AspNetCore.

**Comment by blowdart** [comment]
Darn, was worth a try

**Comment by petersladek** [comment]
Is this on a roadmap or not really? Is there any workaround to get HttpContext.User.Identity populated for in-process hosting model if I am using IIS client certificate mapping?

**Comment by petersladek** [comment]
Given there is going to be fix in .net 10 for ImpersonationToken [Use ImpersonationToken for IIS Windows Auth](https://github.com/dotnet/aspnetcore/pull/58041), is this issue considered to be fixed too? We are migrating our old wcf service to corewcf service, and would like to still rely on IIS client cert mapping to windows users, as we need to do additional authorization checks inside the service.
